## Task 7 — Subscription Plan Distribution

### Business Question

How are customers distributed across subscription plans?

### Analysis

Group customers by subscription plan, calculate the number of customers and their percentage share of the total number of subscriptions, and compare the distribution across plans.

In [2]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
plan_name_count = starlink.groupby("plan_name").agg(
    users_count = ("unique_id", "count")
)
total_users_count = starlink["plan_name"].count()
plan_name_count["share_pct"] = (
    (plan_name_count["users_count"] / total_users_count) * 100.0
)
print(plan_name_count)

             users_count  share_pct
plan_name                          
Business            1019      10.19
Mini                 944       9.44
Residential         5959      59.59
Roam                2078      20.78


### Result

The Residential plan has the largest customer base with 5,959 customers (59.59%), followed by Roam with 2,078 customers (20.78%). Business accounts for 1,019 customers (10.19%), while Mini has the smallest customer base with 944 customers (9.44%).

### Business Insight

The Residential plan dominates the subscription base, accounting for almost 60% of all subscriptions. This high customer share likely contributes to its leading total revenue identified in the previous analysis. Roam represents the second-largest customer segment, while Business and Mini have significantly smaller customer bases.